src/expr/src/linear.rs

```diff
@@ -1642,6 +1642,11 @@ pub mod plan {
                     (Some(lower_bound), None) => {
                         (Some(Ok((row_builder.clone(), lower_bound, diff))), None)
                     }
+                    // NB: as of 2022-03-02 this code is dead, since
```

Comment:
I remember I changed this code to not match on `Option<lower bound>` but rather unwrap in the else branch further up, which would make this code path impossible to reach. I'm wondering if this is actually the right thing to do since we can't get here with a `None` lower bound.

I'm not sure about leaving the code here that is never exercised and where it's not clear what the intention was from just looking at the single line of code. I'd rather have an unwrap earlier in the code, or mark this as `unreachable!`.

Reply:
I didn't see Frank's comment on #10976, so I'd opt to mark this as `unreachable!`.

Reply:
Strong +1 on that, assuming @frankmcsherry is on board.